src/grpc/clients/ea_restaurant_java_etl_grpc_client.py

```diff
@@ -47,7 +49,43 @@ def insert_mongo_order_status_histories_from_python_etl(
                 mongoOrderStatusHistory=mongo_order_status_histories
             )
         )
-        metadata = (("authorization", access_token),)
+        metadata = (
+            ("authorization", generate_bearer_token_from_access_token(access_token)),
+        )
         return self.mongo_order_status_service_stub.insertMongoOrderStatusHistoriesFromPythonEtl(
             mongo_order_status_histories_request, metadata=metadata
         )
+
+    def execute_service_with_oauth2_auth(self, grpc_call, login_client, refresh_token):
+        first_time_login = False
+
+        try:
+            if not first_time_login:
+                login_client()
+                first_time_login = True
+
+            grpc_call()
+
+        except Exception as grpc_call_exception:
+
+            if grpc_call_exception.details() == JwtException.EXPIRED_JWT.value:
```

Comment:
Could you try to check what options do you have here? I don't think comparing string is a good idea at all. Can you think of another way of doing this? the requirement is simple, it should be unique and constant / immutable

Reply:
Done